File: examples-sdk/zig/013-basic-nats-worker/main.zig
// 013-basic-nats-worker — Zig SDK equivalent
// Compile: vil compile --from zig --input 013-basic-nats-worker/main.zig --release

const vil = @import("vil");

pub fn main() void {
    var server = vil.Server.init("nats-worker", 8080);
    var nats = vil.Service.init("nats");
    nats.endpoint("GET", "/nats/config", "nats_config");
    nats.endpoint("POST", "/nats/publish", "nats_publish");
    nats.endpoint("GET", "/nats/jetstream", "jetstream_info");
    nats.endpoint("GET", "/nats/kv", "kv_demo");
    server.service(&nats);
    var root = vil.Service.init("root");
    server.service(&root);
    server.compile();
}

File: examples-sdk/go/013-basic-nats-worker/main.go
// 013-basic-nats-worker — Go SDK equivalent
// Compile: vil compile --from go --input 013-basic-nats-worker/main.go --release
package main

import vil "github.com/OceanOS-id/vil-go"

func main() {
	s := vil.NewServer("nats-worker", 8080)

	nats := vil.NewService("nats")
	nats.Endpoint("GET", "/nats/config", "nats_config")
	nats.Endpoint("POST", "/nats/publish", "nats_publish")
	nats.Endpoint("GET", "/nats/jetstream", "jetstream_info")
	nats.Endpoint("GET", "/nats/kv", "kv_demo")
	s.Service(nats)

	root := vil.NewService("root")
	s.Service(root)

	s.Compile()
}
